### PR TITLE
query params provider and slot filling

### PR DESCRIPTION
**Pull request by morfeusys**
This PR appends two more features to Dialogflow:

* slot filling support
* query parameters provider

### CODE REVIEW

**Review by Denire — APPROVED** [review]

File: activators/dialogflow/src/main/kotlin/com/justai/jaicf/activator/dialogflow/DialogflowIntentActivator.kt [hunk 1]
```diff
@@ -44,9 +39,47 @@ class DialogflowIntentActivator(
...
+        var context = activatorContext?.dialogflow?.apply {
+            if (queryResult.allRequiredParamsPresent) {
+                return SlotFillingSkipped
+            }
+        }
+        if (context == null) {
+            context = recogniseIntent(botContext, request).firstOrNull()
+        }
+        return when {
+            context == null -> SlotFillingInterrupted
```
Comment (nikvoloshin) [inline activators/dialogflow/src/main/kotlin/com/justai/jaicf/activator/dialogflow/DialogflowIntentActivator.kt:58]: It seems like if an intent has required parameters, then on the first "interruption" by user, dialogflow will interrupt slotfilling, but will return the same intent in the response. So, in the current implementation, we will return `SlotFillingInProgress` in this case, as I understand

User: Hello
Dialogflow: "HelloIntent", action = "user_defined_slotfilling_action_name"
Bot: Hi, what is your name? (required) 
User: Cancel
Dialogflow: "HelloIntent", action = null
User: Cancel
Dialogflow: "Fallback"

===================================

Also, we should check `queryResult.isFallback`, because even in the case of fallback Dialogflow will return intent, and context will not be null. I guess, this check should be done in `recognizeIntent`

File: activators/dialogflow/src/main/kotlin/com/justai/jaicf/activator/dialogflow/DialogflowIntentActivator.kt [hunk 2]
```diff
@@ -44,9 +39,47 @@ class DialogflowIntentActivator(
...
+            context.queryResult.allRequiredParamsPresent -> SlotFillingFinished(context)
```
Comment (nikvoloshin) [inline activators/dialogflow/src/main/kotlin/com/justai/jaicf/activator/dialogflow/DialogflowIntentActivator.kt:59]: Shouldn't we respond with Dialogflow prompt message? The same for `SlotFillingSkipped` case.

-> Hello
<- Hi, what is your name?
-> John
<- Nice to meet you, John! (will not be displayed)

**Review by nikvoloshin — CHANGES_REQUESTED** [review]
Also, an important thing not about just slot-filling, but about stateful NLU services at all, one of the reasons why I'm stuck with Lex.  We should somehow manage the outside context. 
The first case (currently not supported), what if we've done `recognizeIntent`, but then decided not to stay with this activator? We should then clean the outside context for the next request.
The second case (more realistic), what if during the slot-filling process user has changed intent? Then we should interrupt slot-filling AND clear the context for the next request.
In other words, every time we are done (either with success or failure) with the current intent in a bot, we should inform NLU.

File: activators/dialogflow/src/main/kotlin/com/justai/jaicf/activator/dialogflow/DialogflowIntentActivator.kt (hunk 1 above)
Comment (morfeusys) [reply activators/dialogflow/src/main/kotlin/com/justai/jaicf/activator/dialogflow/DialogflowIntentActivator.kt:58]: No, in this case we return `SlotFillingFinished`. You can just check the current example using "wake up" input to trigger the slot filling and then send "cancel".

**Review by nikvoloshin — APPROVED** [review]
